### PR TITLE
Add Claude quota management endpoints

### PR DESCRIPTION
**Pull request by mcowger**
Implements management endpoints for querying and refreshing Claude API quota information from OAuth accounts.

## New Endpoints

**GET /v0/management/claude-api-key/quotas**
Returns quota information for all Claude OAuth accounts.

**GET /v0/management/claude-api-key/quota/:authId**
Returns quota information for a specific Claude OAuth account.

**POST /v0/management/claude-api-key/quota/:authId/refresh**
Performs a minimal inference request to refresh quota information for a specific account.

### Quota Data Structure ([`claude_quota.go`](internal/runtime/executor/claude_quota.go))
Defines [`ClaudeCodeQuotaInfo`](internal/runtime/executor/claude_quota.go:11) struct to capture Anthropic rate limit headers:
- Unified status and reset time
- 5-hour window status, reset, and utilization
- 7-day window status, reset, and utilization  
- Overage quota status, reset, and utilization
- Representative claim, fallback percentage, and availability

### Passive Quota Capture ([`claude_executor.go`](internal/runtime/executor/claude_executor.go))
Modified [`Execute`](internal/runtime/executor/claude_executor.go:44) and [`ExecuteStream`](internal/runtime/executor/claude_executor.go:162) to parse quota headers from every Claude API response and store in auth metadata.

### Active Quota Refresh ([`claude_executor.go`](internal/runtime/executor/claude_executor.go:432))
Added [`CheckQuota`](internal/runtime/executor/claude_executor.go:432) method that sends a minimal request (claude-haiku-4-5-20251001, max_tokens=1) to retrieve current quota headers.

### Management Handlers ([`claude_quota.go`](internal/api/handlers/management/claude_quota.go))
- [`GetClaudeCodeQuotas`](internal/api/handlers/management/claude_quota.go:46): Lists all Claude OAuth account quotas
- [`GetClaudeCodeQuota`](internal/api/handlers/management/claude_quota.go:83): Gets quota for specific account
- [`RefreshClaudeCodeQuota`](internal/api/handlers/management/claude_quota.go:127): Triggers quota refresh and persists updated metadata
- [`extractClaudeQuota`](internal/api/handlers/management/claude_quota.go:14): Handles both in-memory pointers and JSON-deserialized maps from disk

### Route Registration ([`server.go`](internal/api/server.go))
Registered new endpoints in management route group.

## Test Coverage

Added comprehensive test suites (964 lines):

**Handler Tests** ([`claude_quota_test.go`](internal/api/handlers/management/claude_quota_test.go)):
- Quota extraction from various metadata formats
- GET/POST endpoint behavior
- Error handling for invalid accounts and missing auth manager
- Defensive metadata initialization

**Executor Tests** ([`claude_quota_test.go`](internal/runtime/executor/claude_quota_test.go)):
- Header parsing for all Anthropic rate limit fields
- Unix timestamp and float parsing edge cases
- Partial and missing header handling

### CODE REVIEW

File: internal/api/handlers/management/claude_quota.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	// This happens when auth is loaded from disk
+	jsonBytes, err := json.Marshal(raw)
+	if err != nil {
+		return nil
+	}
+
+	var quota executor.ClaudeCodeQuotaInfo
+	if err := json.Unmarshal(jsonBytes, &quota); err != nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (mcowger) [reply internal/api/handlers/management/claude_quota.go:77]: It has a log.

**Comment by luispater** [comment]
CLIProxyAPI has no plans to add a quota query feature.

**Comment by mcowger** [comment]
> CLIProxyAPI has no plans to add a quota query feature.

It already has it for 2 other tools...but OK.

**Comment by ThrownLemon** [comment]
> CLIProxyAPI has no plans to add a quota query feature.

Why would you reject this when you already have quotas for other models in the project? this a missing feature that is required to complete an exsiting feature. @luispater can you please recosider this PR? 

Thanks in advance.

Screenshot 2026-01-03 at 2 31 37 pm

**Comment by mcowger** [comment]
This has happened a couple times with this project.  The owner has very strong opinions and isn't super interested in outside ideas. 

I've moved on and built my own (more flexible) replacement.
